website/blog/2022-05-10-faster-presto-queries-with-parquet-page-index.md

```diff
@@ -0,0 +1,54 @@
+
+In the first example below, we have a query which has a filter ‘x = 2000’. For demonstration purposes only, we just show three pages in a column chunk. The three pages in the diagram have statistics that forms ranges of [-100, 60],  [50, 234], [800, 1000]. The column chunk has the statistic that forms the range [-100, 1000]. Since we are looking for the value of 2000, neither page statistics nor the column chunk statistics indicate we should read the data in this column. In this case, page statistics and column chunk statistics equally effective to reduce data reading. 
+
+![Effectiveness Comparison: Both Reject](/img/blog/2022-05-10-faster-presto-queries-with-parquet-page-index/image8.png)
+
+In the second example, the data and statistics in both column chunk and page are the same as the first example. The only change is the filter ‘x = 55’.  In this case, the column chunk statistic range [-100, 1000] determines ‘yes’ to read because it falls into its range. The first page statistic range [50, 234] and the second one [-100, 60]  also determines ‘yes’ for the same reason, but the last page determines ‘no’, because it fall out of it range [800, 1000]
+
+![Effectiveness Comparison: Column Statistics accept, Page Statistics partially accept](/img/blog/2022-05-10-faster-presto-queries-with-parquet-page-index/image5.png)
+
+In the third example, the data and statistics in both column chunk and page are still the same as the first example, but the filter is changed to ‘x =  700’. The column chunk statistic still determines ‘yes’ because it falls into its range [-100, 1000], but all the page statistics determine ‘no’ because none of them have a range that can cover the value ‘700’ that we are looking for. As a result, the whole column data is skipped for reading. 
+
+![Effectiveness Comparison: Column Statistics accept, Page Statistics reject](/img/blog/2022-05-10-faster-presto-queries-with-parquet-page-index/image6.png)
+
+In the above examples, we only show 3 pages in a column chunk, but in reality generally it has tens or hundreds pages in a column chunk. So the savings would be larger in reality. If the column data is sorted, the page statistics work the better because the possibility of false alarm is reduced.   
+
+## Page Statistics in Presto
+The page statistics in older versions than 1.11.0 are placed in the page header. The problem is that the reader has to read each individual page to get the page statistics. Then even it determines later not to read the page, but it is already read. The parquet-mr 1.11.0 fixed this issue by placing all the page statistics for a column chunk into one place, so that the reader can read them at once and make determination which page should be read.
```

Comment:
```suggestion
The page statistics in older versions than 1.11.0 are placed in the page header. The problem is that the reader has to read each individual page to get the page statistics. Then even if it determines later not to read the page, but it is already read. The parquet-mr 1.11.0 fixed this issue by placing all the page statistics for a column chunk into one place, so that the reader can read them at once and make determination which page should be read.
```